bundle-workflow/python/build_workflow/builder.py

```diff
@@ -0,0 +1,48 @@
+# Copyright OpenSearch Contributors.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+
+'''
+This class is responsible for executing the build for a component and passing the results to an output recorder.
+It will notify the output recorder of build information such as repository and git ref, and any artifacts generated by the build.
+Artifacts found in "<build root>/artifacts/<maven|plugins|libs>" will be recognized and recorded.
+'''
+class Builder:
+    def __init__(self, component_name, git_repo, component_scripts_path, default_build_script, output):
+        '''
+        Construct a new Builder instance.
+        :param component_name: The name of the component to build.
+        :param git_repo: A GitRepository instance containing the checked-out code.
+        :param component_scripts_path: Where to look for individual component "build.sh" scripts for those repositories that don't contain their own "scripts/build.sh".
+        :param default_build_script: Path to a defauld build script to use when neither a per-repo "scripts/build.sh" nor a "component_scripts_path/<component>/build.sh" script can be found.
+        :param output: The output recorder that will capture build information and artifacts.
+        '''
+
+        self.component_name = component_name
+        self.git_repo = git_repo
+        self.component_scripts_path = component_scripts_path
+        self.default_build_script = default_build_script
+        self.output = output
+
+    def build(self, version, arch):
+        paths = [os.path.realpath(os.path.join(self.git_repo.dir.name, 'scripts/build.sh')),
+                 os.path.realpath(os.path.join(self.component_scripts_path, self.component_name, 'build.sh')),
+                 self.default_build_script]
+
+        build_script = next(filter(lambda path: os.path.exists(path), paths), None)
+        if build_script is None:
+            raise RuntimeError(f'Could not find build.sh script. Looked in {paths}')
+
+        build_command = f'{build_script} {version} {arch}'
+        self.git_repo.execute(build_command)
+        self.output.record_component(self.component_name, self.git_repo)
+
+    def export_artifacts(self):
+        artifacts_dir = os.path.realpath(os.path.join(self.git_repo.dir.name, 'artifacts'))
+        for artifact_type in ["maven", "plugins", "libs"]:
```

Comment:
this is missing "bundle".  Opensearch core's build script will place the min bundle in that directory.

Reply:
Ah, I missed that. Good catch!